pkg/apis/eventing/v1/trigger_types.go

```diff
@@ -76,8 +76,9 @@ var (
 type TriggerSpec struct {
 	// Broker is the broker that this trigger receives events from.
 	Broker string `json:"broker,omitempty"`
-	// BrokerNamespace is the namespace of the broker that the trigger receives events from.
-	BrokerNamespace string `json:"brokerNamespace,omitempty"`
+
+	// BrokerRef is the broker that is used for cross-namespace referencing.
+	BrokerRef duckv1.KReference `json:"brokerRef,omitempty"`
```

Comment:
I think subscription didn't use a pointer to a channel. Should we keep it consistent as well?

Reply:
This is just internal, I think this should help fix the upgrade tests that are failing. So, for end users this would seem to be the same as the channel